pkg/probe/crd_probe.go

```diff
@@ -58,23 +65,31 @@ var customResourceDefinitionNames = []string{
 	ServiceBinding,
 }
 
-// ReadinessCRD provides functionality that ensures that all ServiceCatalog CRDs are ready
-type ReadinessCRD struct {
-	client apiextensionsclientset.Interface
+// CRDProbe provides functionality that ensures that all ServiceCatalog CRDs are ready
+type CRDProbe struct {
+	client  apiextensionsclientset.Interface
+	delay   int
+	counter int
 }
 
-// NewReadinessCRDProbe returns pointer to ReadinessCRD
-func NewReadinessCRDProbe(apiextensionsClient apiextensionsclientset.Interface) (*ReadinessCRD, error) {
-	return &ReadinessCRD{client: apiextensionsClient}, nil
+// NewCRDProbe returns pointer to CRDProbe
+func NewCRDProbe(apiextensionsClient apiextensionsclientset.Interface, delay int) *CRDProbe {
+	return &CRDProbe{client: apiextensionsClient, counter: 0, delay: delay}
 }
 
-// Name returns name of readiness probe
-func (r ReadinessCRD) Name() string {
-	return "ready-CRDs"
+// Name returns name of CRD probe
+func (r CRDProbe) Name() string {
+	return fmt.Sprintf("ready-CRDs-%d", rand.Intn(1000))
 }
 
 // Check if all CRDs with specific label are ready
-func (r *ReadinessCRD) Check(_ *http.Request) error {
+func (r *CRDProbe) Check(_ *http.Request) error {
+	if r.counter < r.delay {
+		r.counter++
+		klog.V(4).Infof("CRDProbe %s skiped. Ckeck for %d iteration(s)", r.Name(), r.delay-r.counter)
```

Comment:
maybe sth like
```
Will be executed in %d iteration
```

@klaudiagrz please check that also:)

Reply:
```suggestion
		klog.V(4).Infof("%s CRDProbe skipped. Will be executed in %d iteration", r.Name(), r.delay-r.counter)
```